migration/v0.43/apply/20-upgrade-opensearch.sh

```diff
@@ -0,0 +1,155 @@
+
+          log_info "- Getting name of .opensearch_dashboards index"
+          os_dashboards_index=$(curl -sS -kL -u "${user}:${password}" -X GET "${os_url}"/_alias/.opensearch_dashboards | jq -r 'to_entries | .[0].key')
+
+          if [[ "${os_dashboards_index}" != .opensearch_dashboards* ]]; then
+            log_fatal "Failed to get index name of the .opensearch_dashboards alias"
+          fi
+
+          log_info "- Marking index '${os_dashboards_index}' as read-only"
+          resp=$(curl -sS -kL -u "${user}:${password}" -X PUT "${os_url}"/"${os_dashboards_index}"/_settings -H 'Content-Type: application/json' -d'
+          {
+            "settings": {
+              "index.blocks.write": true
+            }
+          }
+          ')
+
+          acknowledged=$(echo "${resp}" | grep "^{" | jq -r '.acknowledged')
+          if [ "${acknowledged}" = "true" ]; then
+            log_info "- Marked '${os_dashboards_index}' as read-only"
+          else
+            log_fatal "Failed to mark index '${os_dashboards_index}' as read-only" "${resp}"
+          fi
+
+          log_info "- Cloning index '${os_dashboards_index}' to index .kibana_1"
+          resp=$(curl -sS -kL -u "${user}:${password}" -X PUT "${os_url}"/"${os_dashboards_index}"/_clone/.kibana_1)
+          acknowledged=$(echo "${resp}" | grep "^{" | jq -r '.acknowledged')
+          if [ "${acknowledged}" = "true" ]; then
+            log_info "- Successfully cloned '${os_dashboards_index}' to .kibana_1"
+          else
+            log_fatal "Failed to clone index '${os_dashboards_index}' to .kibana_1" "${resp}"
+          fi
+
+
+          log_info "- Disabling read-only mode for '${os_dashboards_index}'"
+          resp=$(curl -sS -kL -u "${user}:${password}" -X PUT "${os_url}"/"${os_dashboards_index}"/_settings -H 'Content-Type: application/json' -d'
+          {
+            "settings": {
+              "index.blocks.write": false
+            }
+          }
+          ')
+          acknowledged=$(echo "${resp}" | grep "^{" | jq -r '.acknowledged')
+          if [ "${acknowledged}" = "true" ]; then
+            log_info "- Successfully disabled read-only mode for '${os_dashboards_index}'"
+          else
+            log_fatal "Failed to disable read-only mode for '${os_dashboards_index}'" "${resp}"
+          fi
+
+          log_info "- Disabling read-only mode for .kibana_1"
+          resp=$(curl -sS -kL -u "${user}:${password}" -X PUT "${os_url}"/.kibana_1/_settings -H 'Content-Type: application/json' -d'
+          {
+            "settings": {
+              "index.blocks.write": false
+            }
+          }
+          ')
+          acknowledged=$(echo "${resp}" | grep "^{" | jq -r '.acknowledged')
+          if [ "${acknowledged}" = "true" ]; then
+            log_info "- Successfully disabled read-only mode for .kibana_1"
+          else
+            log_fatal "Failed to disable read-only mode for .kibana_1" "${resp}"
+          fi
```

Comment:
Maybe this should be outside of the if index exist, in case fails between creation and setting read-write?
Also, do you need to set the `.opensearch_dashboards` index as read-write if you are going to remove it?

Reply:
No it's not necessary for the `.opensearch_dashboards` index, I was going to try removing that part but forgot :sweat_smile: 

And yeah I think it makes sense to move this outside the if-statement.